Add k8s-1.35 and ecr-credential-provider-1.35 pkgs with pre-release sources

ytsssun opened the pull request:
**Issue number:**

Related to: https://github.com/bottlerocket-os/bottlerocket/issues/4725

**Description of changes:**

Initial set of changes for kubernetes 1.35 support:
- Add a kubernetes-1.35 pkg with the EKS-D k8s-1.35 beta source (v1.35.0-beta.0)
- Add a ecr-credential-provider-1.35 pkg with the 1.35.0 source

Key changes:
- Remove `--pod-infra-container-image` kubelet flag (removed upstream in 1.35)

**Testing done:**
- [x] Core-kit builds with the new packages
- [x] Instance using the 1.35-beta.0 kubelet boots
   **kubelet version is reported correctly**
   ```
    bash-5.1# kubelet --version
    Kubernetes v1.35.0-beta.0-eks-7980c10
   ```
   **Node joined cluster**
   ```
   k get nodes
   NAME                                            STATUS   ROLES    AGE     VERSION
   ip-192-168-2-215.us-west-2.compute.internal     Ready    <none>   2m3s    v1.35.0-beta.0-eks-7980c10
   ```
- [x] RPM info to verify `rpmvercmp`
```
rpm -qpi build/rpms/kubernetes-1.35/bottlerocket-kubelet-1.35-1.35.0-0.beta0.1767128447.473aa566.br1.x86_64.rpm 
Name        : bottlerocket-kubelet-1.35
Version     : 1.35.0
Release     : 0.beta0.1767128447.473aa566.br1
Architecture: x86_64
Install Date: (not installed)
Group       : Unspecified
Size        : 1579759
License     : Apache-2.0 AND BSD-3-Clause
Signature   : (none)
Source RPM  : bottlerocket-kubernetes-1.35.0-0.beta0.1767128447.473aa566.br1.src.rpm
Build Date  : Tue 30 Dec 2025 09:07:20 PM UTC
Build Host  : localhost
URL         : https://github.com/kubernetes/kubernetes
Summary     : Container cluster node agent
Description :
Container cluster node agent.

rpm -qpi build/rpms/ecr-credential-provider-1.35/bottlerocket-ecr-credential-provider-1.35-1.35.0-1.1767144308.398157c0.br1.x86_64.rpm 
Name        : bottlerocket-ecr-credential-provider-1.35
Version     : 1.35.0
Release     : 1.1767144308.398157c0.br1
Architecture: x86_64
Install Date: (not installed)
Group       : Unspecified
Size        : 1095302
License     : Apache-2.0
Signature   : (none)
Source RPM  : bottlerocket-ecr-credential-provider-1.35-1.35.0-1.1767144308.398157c0.br1.src.rpm
Build Date  : Wed 31 Dec 2025 01:25:48 AM UTC
Build Host  : localhost
URL         : https://github.com/kubernetes/cloud-provider-aws
Summary     : Amazon ECR credential provider
Description :
Amazon ECR credential provider.
```
**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

koooosh commented on packages/ecr-credential-provider-1.35/ecr-credential-provider-1.35.spec:
```diff
@@ -0,0 +1,83 @@
+%global goimport %{goproject}/%{gorepo}
+
+%global gover 1.35.0-beta.0
+%global rpmver 1.35.0
```
Are we using the official 1.35.0 source for cloud-provider? If yes, we can simplify this to:
```
%global gover 1.35.0
%global rpmver %{gover}
```
as we don't need the beta suffix.

If not, then we need to edit the `Release` field below to indicate this is a pre-release: `0.beta0%{?dist}`

ytsssun replied:
Updated thanks.

koooosh commented on packages/ecr-credential-provider-1.35/ecr-credential-provider-1.35.spec:
```diff
@@ -0,0 +1,83 @@
+%set_cross_go_flags
+
+export GOTOOLCHAIN=local
+export GO_MAJOR="1.24"
```
The upstream source uses Go 1.25: https://github.com/kubernetes/cloud-provider-aws/blob/v1.35.0/go.mod#L3

ytsssun replied:
Yeah, the change that bump the version went to the wrong commit. This will be addressed.

koooosh commented on packages/ecr-credential-provider-1.35/ecr-credential-provider-1.35.spec:
```diff
@@ -0,0 +1,83 @@
+Source1: bundled-cloud-provider-aws-%{gover}.tar.gz
+Source1000: clarify.toml
+
+Patch0001: 0001-support-new-aws-partition-in-credential-provider.patch
```
According to [the 1.35.0 release notes](https://github.com/kubernetes/cloud-provider-aws/releases/tag/v1.35.0), this fix was merged upstream, so we shouldn't need to carry this patch: https://github.com/kubernetes/cloud-provider-aws/pull/1284

ytsssun replied:
Updated thanks. Can confirm that is merged upstream.

koooosh commented on packages/ecr-credential-provider-1.35/ecr-credential-provider-1.35.spec:
```diff
@@ -0,0 +1,83 @@
+Source1000: clarify.toml
+
+Patch0001: 0001-support-new-aws-partition-in-credential-provider.patch
+Patch0002: 0002-ecr-credential-provider-hardcode-ECR-endpoint-for-eu.patch
```
I don't see this patch merged upstream: https://github.com/kubernetes/cloud-provider-aws/blob/v1.35.0/cmd/ecr-credential-provider/main.go

But we should verify with @sam-berning if we still need to carry this for 1.35

ytsssun replied:
Checked with @sam-berning , will keep this patch for now. We need to confirm that the new go sdk has the endpoint for the eu region before dropping it. And we should test this endpoint is reachable before dropping this patch.

koooosh commented on packages/ecr-credential-provider-1.35/ecr-credential-provider-1.35.spec:
```diff
@@ -2,7 +2,7 @@
 %global goimport %{goproject}/%{gorepo}
 
-%global gover 1.35.0-beta.0
+%global gover 1.35.0
```
I left a comment on the ecr commit with a similar fix, but this should go in the other commit

koooosh commented on packages/kubernetes-1.35/kubernetes-1.35.spec:
```diff
@@ -0,0 +1,283 @@
+
+%global releasever 1
+%global gover 1.35.0-beta.0
+%global rpmver 1.35.0~beta.0
```
We prefer to indicate that this is pre-release in the `Release` field rather than `Version`, so we can change these to:
```
%global gover 1.35.0
%global rpmver %{gover}
```

Due to this, we will need to hardcode `Source0` for now, but that's okay as we can update this when adding the official source.

koooosh commented on packages/kubernetes-1.35/kubernetes-1.35.spec:
```diff
@@ -0,0 +1,283 @@
+
+Name: %{_cross_os}%{gorepo}
+Version: %{rpmver}
+Release: 1%{?dist}
```
We must change this to `Release: 0.beta0%{?dist}` to prevent RPM versioning conflicts

ytsssun replied:
Nice catch, updated.

koooosh commented on packages/kubernetes-1.35/pod-infra-container-image:
```diff
@@ -0,0 +1,6 @@
+kubernetes = { version = "v1", optional = true }
++++
+{{~#if settings.kubernetes.pod-infra-container-image~}}
+DEPRECATED_SETTING=settings.kubernetes.pod-infra-container-image
```
Given: `Remove --pod-infra-container-image kubelet flag (removed upstream in 1.35)` do we still need this file?

ytsssun replied:
We actually need this file. It is different from the dropped flag `--pod-infra-container-image` on kubelet.

This file is to indicate that the bottlerocket setting `settings.kubernetes.pod-infra-container-image` is deprecated and will no longer have any effect. Right now this setting is still in the API model, user can still set it, but it is just no-op.

ytsssun committed:
packages: add ecr-credential-provider-1.35

Add ecr-credential-provider for Kubernetes 1.35.
Uses cloud-provider-aws v1.35.0 source.

Signed-off-by: Yutong Sun <yutongsu@amazon.com>

ytsssun committed:
packages: add kubernetes-1.35

Signed-off-by: Yutong Sun <yutongsu@amazon.com>

koooosh commented on packages/kubernetes-1.35/kubernetes-1.35.spec:
```diff
@@ -0,0 +1,283 @@
+%global gorepo kubernetes
+%global goimport %{goproject}/%{gorepo}
+
+%global releasever 0
```
nit: This should be `1`. It corresponds to the `1` in Source0: `https://distro.eks.amazonaws.com/kubernetes-1-35/releases/1 ...`

koooosh commented on packages/kubernetes-1.35/kubernetes-1.35.spec:
```diff
@@ -0,0 +1,283 @@
+Source8: kubernetes-tmpfiles.conf
+Source9: kubelet-sysctl.conf
+Source10: prepare-var-lib-kubelet.service
+Source102:      pod-infra-container-image
```
This is out of place, looks like the most recent revision moved lines related to pod-infra-container-image

koooosh commented on packages/ecr-credential-provider-1.35/ecr-credential-provider-1.35.spec:
```diff
@@ -0,0 +1,82 @@
+
+Name: %{_cross_os}ecr-credential-provider-1.35
+Version: %{rpmver}
+Release: 0.beta0%{?dist}
```
Given this is the official 1.35.0 source for cloud provider (AFAIK), we should have this as: `1%{?dist}`

koooosh reviewed (changes requested):
We should fix the `Release` field for ECR cred provider, but the rest are just nits

ytsssun commented:
Force push to address comments from @koooosh

koooosh commented on packages/kubernetes-1.35/kubernetes-1.35.spec:
```diff
@@ -0,0 +1,283 @@
+# base Apache-2.0, third_party Apache-2.0 AND BSD-3-Clause
+License: Apache-2.0 AND BSD-3-Clause
+URL: https://%{goimport}
+Source0: https://distro.eks.amazonaws.com/kubernetes-1-35/releases/%{releasever}/artifacts/kubernetes/v%{gover}/kubernetes-src.tar.gz
```
Shouldn't this be: `https://distro.eks.amazonaws.com/kubernetes-1-35/releases/%{releasever}/artifacts/kubernetes/v1.35.0-beta.0/kubernetes-src.tar.gz` ?

I'm surprised `v1.35.0` works instead of `v1.35.0-beta.0`

ytsssun replied:
Synced offline. This worked because Twoliter would use look-aside-cache for the fetching and conditionally fallback to the `url` specified in `Cargo.toml` (if we specified `UPSTREAM_SOURCE_FALLBACK` for the build).

Quoting from [Maximum RPM](http://ftp.rpm.org/max-rpm/s1-rpm-inside-tags.html#:~:text=Given%20this%20source%20line%2C%20RPM%20will%20search%20its%20SOURCES%20directory%20for%20cdplayer%2D1%2E0%2Etgz%2E%20Everything%20prior%20to%20the%20filename%20is%20ignored%20by%20RPM%2E%20It%27s%20there%20strictly%20for%20any%20interested%20humans%2E): 

>Given this **source** line, RPM will search its `SOURCES` directory for `cdplayer-1.0.tgz`. Everything prior to the filename is ignored by RPM. It's there strictly for any interested humans.


So a plain filename would just suffice here. Given that we constantly update the source url, it is probably better that we keep Source0 with just he filename - to avoid repeated updates.

koooosh reviewed (approved):
The rpm info for `ecr-credential-provider-1.35` needs to be updated in the description since we updated some of the fields, but overall LGTM.
